libraries/AP_WindVane/AP_WindVane.cpp

```diff
@@ -23,11 +23,6 @@
 #include "AP_WindVane_SITL.h"
 #include "AP_WindVane_NMEA.h"
 
-#define WINDVANE_DEFAULT_PIN 15                     // default wind vane sensor analog pin
```

Comment:
Normally we like to keep the definitions in the .cpp files because it has some effect on compile time.  @OXINARF knows the reason better than I do.

Reply:
I'll try to explain:

- compilation is done individually on each .cpp file
- .cpp files include headers (and headers also include other headers)
- if an included header changes, then the .cpp file also changed, so it needs to be recompiled

Giving the example here:
- AP_WindVane.h is included in Rover.h
- Rover.h is included in nearly all (if not all) .cpp files in the APMrover2 folder
- this means that if any of the definitions is changed in AP_WindVane.h then all those .cpp files in APMrover2 need to be recompiled

There is no reason for that because these values are only used in AP_WindVane.cpp and that should be the only file needing recompilation.

Reply:
I think I understand, TACK_FILT_CUTOFF is the only define needed in AP_WindVane.h, I just moved all the rest so they would all be in the same place, I could move the rest back to the cpp?